tensorflow_io/core/plugins/s3/s3_filesystem.cc

```diff
@@ -62,29 +62,61 @@ constexpr size_t kUploadRetries = 3;
 
 constexpr size_t kS3ReadAppendableFileBufferSize = 1024 * 1024;  // 1 MB
 
-static void* plugin_memory_allocate(size_t size) { return calloc(1, size); }
-static void plugin_memory_free(void* ptr) { free(ptr); }
+static inline char* TF_MakeStatusMessage(const Aws::String& str) {
+  char* msg = new char[str.length() + 1];
+  strcpy(msg, str.c_str());
```

Comment:
@vnvo2409 Will that cause memory leak as I don't see they are being freed?

Reply:
[`Status`](https://github.com/tensorflow/tensorflow/blob/fbcb024d7f107700a35226cd83f7ef25a476036e/tensorflow/core/platform/status.cc#L92-L101) will take the ownership of that `char*` so there won't be any leak.

Reply:
@vnvo2409 Inside `Status` a copy is made:

https://github.com/tensorflow/tensorflow/blob/fbcb024d7f107700a35226cd83f7ef25a476036e/tensorflow/core/platform/status.cc#L97

`string` will take the input of `const char *` or `char *` and make a copy internally. I think it does not take ownership. The input will not be modified and it is up to the caller to free.

The `c_str()` returns a `const char*` which will be deleted when it is out of scope, but once the `const char*` passed to `TF_SetStatus` and returned from `TF_SetStatus`, `TF_SetStatus`  will not touch passed `const char*` anymore (as `TF_SetStatus` make a copy).

Can you point me to the place of the issue you were encountered?

Reply:
If so, I am just overthinking about it. Thank you !